.github/workflows/nextjs_bundle_analysis.yml

```
@@ -0,0 +1,117 @@
+name: 'Next.js Bundle Analysis'
+
+on:
+   pull_request:
+   push:
+      branches:
+         - main # change this if your default branch is named differently
+   workflow_dispatch:
+
+defaults:
+   run:
+      # change this if your nextjs app does not live at the root of the repo
+      working-directory: ./frontend
+
+jobs:
+   analyze:
+      runs-on: ubuntu-latest
+      steps:
+         - name: Checkout
+           uses: actions/checkout@v2
+
+         - name: Setup and cache pnpm
+           uses: ./.github/actions/pnpm-setup
+
+         - name: pnpm install
+           run: pnpm install
+           env:
+              FONT_AWESOME_NPM_TOKEN: ${{ secrets.FONT_AWESOME_NPM_TOKEN }}
+
+         - name: Restore next build
+           uses: actions/cache@v2
+           id: restore-build-cache
+           env:
+              cache-name: cache-next-build
+           with:
+              # if you use a custom build directory, replace all instances of `.next` in this file with your build directory
+              # ex: if your app builds to `dist`, replace `.next` with `dist`
+              path: .next/cache
+              # change this if you prefer a more strict cache
+              key: ${{ runner.os }}-build-${{ env.cache-name }}
+
+         - name: Build next.js app
+           # change this if your site requires a custom build command
+           run: ./node_modules/.bin/next build
```

Probably not necessary since we use `next build` in our package.json script, but this is simpler and will automatically pick up changes if we edit our build command down the line.

```suggestion
           run: pnpm build
```

---

I intentionally didn't use that command because this task is explicitly about building next.js and if we change that command to do additional tasks (not sure what they would be) they may not be relevant. That said, I'm not sure which is more likely: changing our setup in a way that `next build` doesn't work, or changing that command in a way that breaks this analysis.

---

Yea good points. I guess I was thinking "we should analyze the bundle the exact same way we build the app for production", but it's hard to imagine how we would augment that command to affect the bundle size. This is fine as is.